dca/methods_comparison.py

```diff
@@ -610,3 +610,344 @@ def fit_transform(self, X, n_components=None):
+        Whether to subtract the cross-condition mean from each condition
+        before running jPCA.
+
+    Attributes
+    ----------
+    eigen_vecs_ : list
+        List of numpy eigenvectors from JPCA skew symmetric matrix sorted in
+        descending order by magnitude of eigenvalue.
+
+    eigen_vals_ : list
+        List of eigenvalues from JPCA skew symmetric matrix. The index
+        of each eigenvalue corresponds with the eigenvector in eigen_vecs_.
+
+    pca_ : sklearn.decomp.PCA object
+        PCA object used to transform X to X_red.
+
+    cross_condition_mean_ : ndarray (time, features)
+        Cross condition mean of X during fit.
+
+    """
+    def __init__(self, n_components=6, mean_subtract=True):
+        if n_components // 2 != n_components / 2:
+            raise ValueError("n_components must be even int")
+        self.n_components_ = n_components
+        self.mean_subtract_ = mean_subtract
+        self.eigen_vecs_ = None
+        self.eigen_vals_ = None
+        self.pca_ = None
+        self.cross_condition_mean_ = None
+
+    def fit(self, X):
+        """ Fit a jPCA model to X.
+
+        Parameters
+        ----------
+        X : ndarray (time, features) or (conditions, time, features)
+            Data to fit using jPCA model.
+
+        Returns
+        -------
+        self
+        """
+        if self.n_components_ > X.shape[1]:
+            raise ValueError("n_components is greater than number of features in X.")
+
+        if self.mean_subtract_:
+            self.cross_condition_mean_ = np.mean(X, axis=0)
+            X = self._cross_condition_subtract(X, self.cross_condition_mean_)
+
+        condition_indices = self._get_condition_indices(X)
+
+        X = np.vstack(X)
+        self.pca_ = PCA(n_components=self.n_components_)
+        self.pca_.fit(X)
+
+        X_red = self.pca_.transform(X)
+
+        dX, X_prestate = self._get_dX_and_prestate(X_red, condition_indices)
+
+        M_skew = self._fit_skew(X_prestate, dX)
+        self.eigen_vals_, self.eigen_vecs_ = self._get_jpcs(M_skew)
+
+        return self
+
+    def transform(self, X):
+        """ Transform X using JPCA components.
+
+        Parameters
+        ----------
+        X : ndarray (time, features) or (conditions, time, features)
+            Data to fit using jPCA model.
+
+        Returns
+        -------
+        ndarray (time*conditions, n_components)
+            X projected onto jPCA components. In X_proj, every pair
+            of features correspond to a conjugate pair of JPCA eigenvectors.
+            The pairs are sorted by largest magnitude eigenvalue
+            (i.e. dimensions 0 and 1 in X_proj contains the projection from
+            the conjugate eigenvector pair with the largest eigenvalue magnitude).
+            The projection pair is what captures the rotations.
+
+        """
+
+        if self.mean_subtract_:
+            X = self._cross_condition_subtract(X, self.cross_condition_mean_)
```

Comment:
You can do this mean subtraction in one line, same as above. No need for a function.